storage/replica_raftstorage.go

```diff
@@ -31,15 +31,15 @@ import (
 )
 
 // InitialState implements the raft.Storage interface.
+// InitialState requires that the replica lock be held.
```

Comment:
This is subtle and deserves an extra comment. `InitialState` is called by raft, so this requirement implies that all calls to `raft.RawNode` methods must hold the replica lock.